### DISCUSSION TITLE
Automatically re-connect if connection lost

### DISCUSSION THREAD

**Discussion by ivanperez-keera** [jaxa/racs2_bridge#14] [Ideas]
When testing applications using the bridge, it's often the case that one needs to re-start the ROS applications, or the cFS applications.

It'd be very useful to be able to restart only one end and have the other re-connect automatically.

**Comment by tt-saito** [comment]
Currently, the bridge node on the ROS2 side communicates as a server and the bridge application on the cFS side communicates as a client.

Therefore, as long as the bridge node on the ROS2 side is running, the connection is automatically resumed even if the application on the cFS side is stopped and restarted.

On the other hand, once the bridge node on the ROS2 side is stopped and restarted, the application on the cFS side must also be restarted.

I certainly hope that this can be improved.
